docker/sagemaker/Dockerfile.sm

```diff
@@ -42,6 +31,18 @@ RUN apt-get update; apt-get install -y --no-install-recommends libopenmpi-dev \
 COPY build_artifacts/changehostname.c /opt/ml/code/
 COPY build_artifacts/start_with_right_hostname.sh /usr/local/bin/start_with_right_hostname.sh
 
+FROM base as base-cpu
+
+# Install DGL CPU version
+RUN pip3 install dgl==${DGL_VERSION} -f https://data.dgl.ai/wheels/torch-2.3/repo.html && rm -rf /root/.cache
+
+FROM base as base-gpu
+
+# Install DGL GPU version
+RUN pip3 install dgl==${DGL_VERSION}+cu121 -f https://data.dgl.ai/wheels/torch-2.3/cu121/repo.html && rm -rf /root/.cache
```

Comment:
cuda version here should rely on the base image, if we allow the input image as an argument we would better make here an argument as well.

Reply:
We could grab the CUDA version from the source image. The argument is not exposed to the user, we only use it as a way to make building the images more customizable. Let me see if there's a good way to do this.